pkg/river/vm/constant.go

```diff
@@ -18,11 +18,22 @@ func valueFromLiteral(lit string, tok token.Token) (value.Value, error) {
 		return value.Null, nil
 
 	case token.NUMBER:
-		v, err := strconv.ParseInt(lit, 0, 64)
-		if err != nil {
-			return value.Null, err
+		intVal, err1 := strconv.ParseInt(lit, 0, 64)
+		if err1 == nil {
+			return value.Int(intVal), nil
+		}
+
+		uintVal, err2 := strconv.ParseUint(lit, 0, 64)
+		if err2 == nil {
+			return value.Uint(uintVal), nil
 		}
-		return value.Int(v), nil
+
+		floatVal, err3 := strconv.ParseFloat(lit, 64)
+		if err3 == nil {
+			return value.Float(floatVal), nil
+		}
+
+		return value.Null, err3
```

Comment:
Can you write a benchmark for this and provide before/after results? This may make decoding floating point numbers significantly slower.

Reply:
What triggers calling this codepath? I wonder how much of a hot path this is.

Reply:
It's any time a number is in the config file and a component is evaluated, so it can happen multiple times a second

Reply:
Paulin and I wrote benchmarks together, but it's not possible to compare to the existing code since the existing code forced number literals to fit within an int64. 

The new code is definitely slower in the worst case (float64), where it has to parse the same string three times, but the worst case is still roughly 415ns/op on my machine, which is roughly twice as slow as the best case in main (239ns/op). 

I don't think users are usually going to have number literals which are so big that they fall into the worse case, so I think this change is roughly safe to make without much performance impact; if we do find it to be a performance bottleneck, we can revisit how the numbers are parsed to optimize it.